Realm/RLMSyncConfiguration.h

```diff
@@ -50,7 +50,18 @@ typedef NS_ENUM(NSUInteger, RLMClientResetMode) {
     /// If RLMClientResetModeDiscardLocal is enabled but the client reset operation is unable to complete
     /// then the client reset process reverts to manual mode. Example) During a destructive schema change this
     /// mode will fail and invoke the manual client reset handler.
-    RLMClientResetModeDiscardLocal
+    RLMClientResetModeDiscardLocal,
+    /// The client device will download a realm with objects reflecting the latest version of the server. A recovery
+    /// process is run locally in an attempt to integrate the server version with any local changes from before the
+    /// client reset occurred.
+    /// The changes are integrated with the following rules:
+    /// 1. Objects created locally that were not integrated before client reset will be integrated.
+    /// 2. If an object has been deleted on the server, but was modified on the recovering client, the delete takes precedence and the update is discarded
+    /// 3. If an object was deleted on the recovering client, but not the server, then the client delete instruction is applied.
+    /// 4. In the case of conflicting updates to the same field, the most recent update is applied.
+    RLMClientResetModeRecover,
+    // TODO: docs
```

Comment:
Docs. Need to make a note that .recoverOrDiscardLocal falls back to discard local if the recovery process fails. And .recover falls back to manual client reset handler.